tests/pods_test.go

```diff
@@ -61,36 +69,42 @@ func TestPod(testT *testing.T) {
 		// Verify the container count.
 		require.Equal(retryT, int32(2), pod.ContainerCount)
 
-		// Verify the events.
-		var loopCount int
-		var events []Event
-		for {
-			events = getEvents(retryT, pod)
-			log.Infof("%d: Events: %+v", loopCount, events)
-			if len(events) == 4 {
-				break
+		if os.Getenv("COLLECTION_METHOD") == "NO_COLLECTION" {
+			log.Infof("Skipping parts of TestPod that relate to events because env var \"COLLECTION_METHOD\" is " +
+				"set to \"NO_COLLECTION\"\n. This is expected to only happen when Sensor version is 3.74.x " +
+				"(support exception / compatibility tests)\"")
+		} else {
+			// Verify the events.
+			var loopCount int
+			var events []Event
+			for {
+				events = getEvents(retryT, pod)
+				log.Infof("%d: Events: %+v", loopCount, events)
+				if len(events) == 4 {
+					break
+				}
+				loopCount++
+				require.LessOrEqual(retryT, loopCount, 20)
+				time.Sleep(4 * time.Second)
 			}
-			loopCount++
-			require.LessOrEqual(retryT, loopCount, 20)
-			time.Sleep(4 * time.Second)
-		}
 
-		// Expecting processes: nginx, sh, date, sleep
-		eventNames := sliceutils.Map(events, func(event Event) string { return event.Name })
-		expected := []string{"/bin/date", "/bin/sh", "/bin/sleep", "/usr/sbin/nginx"}
+			// Expecting processes: nginx, sh, date, sleep
+			eventNames := sliceutils.Map(events, func(event Event) string { return event.Name })
+			expected := []string{"/bin/date", "/bin/sh", "/bin/sleep", "/usr/sbin/nginx"}
 
-		log.Infof("Event names: %+v", eventNames)
-		log.Infof("Expected name: %+v", expected)
-		require.ElementsMatch(retryT, eventNames, expected)
+			log.Infof("Event names: %+v", eventNames)
+			log.Infof("Expected name: %+v", expected)
+			require.ElementsMatch(retryT, eventNames, expected)
 
-		// Verify the pod's timestamp is no later than the timestamp of the earliest event.
-		log.Infof("Pod start comparison: %s vs %s", pod.Started, events[0].Timestamp.Time)
-		require.False(retryT, pod.Started.After(events[0].Timestamp.Time))
+			// Verify the pod's timestamp is no later than the timestamp of the earliest event.
+			log.Infof("Pod start comparison: %s vs %s", pod.Started, events[0].Timestamp.Time)
```

Comment:
Those `log.Infof` should be replaced with the test logger: `testT.Logf`. Otherwise the output of logs may not follow the order of things happening in the cluster.

Reply:
Interesting, is this generally the case? Why can we not expect log.Infof to arrive in order?

I also just had a look through the other go e2e tests and all of them are using log.Infof liberally instead of some t.Logf, perhaps they all need to be changed too?

For now I've only changed the ones in PodsTest, the rest could be a follow-up if it's an issue

Reply:
Yes, I think it is generally true that there is no guarantee for order when we mix test and non-test loggers.

All `log.Infof` arrive in order and all `testT.Logf` arrive in order, but the order is kept only within the logger.
So in this synthetic (and a bit exaggerated) example:

```go
func (t *testing.T) TestExample() {
	log.Info("info 1")
	t.Log("test 1")
	log.Info("info 2")
	t.Log("test 2")
}
```

we may see:

```
test 1
test 2
info 1
info 2
```

and theoretically all is good, because `info 1` is before `info 2` and `test 1` is before `test 2`, but it is not what we want, cause we expect rather `info 1, test 1, info 2, test 2`.

Reply:
> I also just had a look through the other go e2e tests and all of them are using log.Infof liberally instead of some t.Logf, perhaps they all need to be changed too?

Definitely, we should use the logger from the `testing.T` inside the tests. 

Using `log.Info` would be okay if we would create the `log` logger by using the logger from the testing library. That would work and the order of messages would be fine, but it would be a bit confusing when reading the code.